Commit 4fd34e28eaf1 broke out-of-source build

Commit 4fd34e28eaf13b0cd6d6dcbcb105be14bccb12e7 broke out-of-source builds. I'm getting:
``` make
Making all in unittests
make[4]: Entering directory '/home/m/nobak/src/OpenSC/b/src/tests/unittests'
make[4]: *** No rule to make target '../../../../src/libopensc/libopensc.la', needed by 'asn1'.  Stop.
make[4]: Leaving directory '/home/m/nobak/src/OpenSC/b/src/tests/unittests'
make[3]: *** [Makefile:640: all-recursive] Error 1
```
Looking at the commit it seems a number of `top_builddir`s were replaced with `top_srcdir`s, causing the problem since `libopensc.la` is a built artifact and no source. This change fixes it for me:
``` diff
diff --git a/src/tests/unittests/Makefile.am b/src/tests/unittests/Makefile.am
index 0795b62a..6d8682b6 100644
--- a/src/tests/unittests/Makefile.am
+++ b/src/tests/unittests/Makefile.am
@@ -16,7 +16,7 @@ AM_CFLAGS = -I$(top_srcdir)/src/ \
        $(CODE_COVERAGE_CFLAGS) \
        $(CMOCKA_CFLAGS)
 AM_CPPFLAGS =$(CODE_COVERAGE_CPPFLAGS)
-LDADD = $(top_srcdir)/src/libopensc/libopensc.la \
+LDADD = $(top_builddir)/src/libopensc/libopensc.la \
        $(CODE_COVERAGE_LIBS) \
        $(OPTIONAL_OPENSSL_LIBS) \
        $(CMOCKA_LIBS)
```
There's another `-L$(top_srcdir)` in `AM_CFLAGS` I can't tell the significance of. It doesn't influence successful build but is also unlikely to work either.

But that's taken care of by `libopensc.la` in `LDADD`, isn't it? Otherwise it wouldn't link and run. `-I$(top_srcdir)/src/libopensc` in `AM_CFLAGS` likely makes the headers available. I removed `-L` from `AM_CFLAGS` without problems. Switching into `src/tests/unittest` runnng `make check` I get:
``` make
make check
make  check-TESTS
make[1]: Entering directory '/build/src/tests/unittests'
make[2]: Entering directory '/build/src/tests/unittests'
PASS: asn1
PASS: simpletlv
PASS: compression
============================================================================
Testsuite summary for OpenSC 0.20.0
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[2]: Leaving directory '/build/src/tests/unittests'
make[1]: Leaving directory '/build/src/tests/unittests'
```

Followup issue: Is `make check` supposed to work from the top source or build dir? With my out-of-source-build I'm now getting:
``` make
make check
Making check in etc
make[1]: Entering directory '/build/etc'
make[1]: Leaving directory '/build/etc'
Making check in src
make[1]: Entering directory '/build/src'
Making check in common
make[2]: Entering directory '/build/src/common'
make  check-local
make[3]: Entering directory '/build/src/common'
if [ -x "/usr/lib/llvm/10/bin/clang-tidy" ]; then clang-tidy -config='' -header-filter=.* compat_dummy.c compat_strlcat.h compat_strlcat.c compat_strlcpy.h compat_strlcpy.c compat_strnlen.h compat_strnlen.c compat_getpass.h compat_getpass.c compat_getopt.h compat_getopt.c compat_report_rangecheckfailure.c compat___iob_func.c simclist.c simclist.h libpkcs11.c libscdl.c -- -DHAVE_CONFIG_H -I. -I../../../src/common -I../..  -I../../../src   -Wall -Wextra -Wno-unused-parameter -Werror -g -O2; fi
Error while processing /build/src/common/compat_dummy.c.
Error while processing /build/src/common/compat_strlcat.h.
Error while processing /build/src/common/compat_strlcat.c.
Error while processing /build/src/common/compat_strlcpy.h.
Error while processing /build/src/common/compat_strlcpy.c.
Error while processing /build/src/common/compat_strnlen.h.
Error while processing /build/src/common/compat_strnlen.c.
Error while processing /build/src/common/compat_getpass.h.
Error while processing /build/src/common/compat_getpass.c.
Error while processing /build/src/common/compat_getopt.h.
Error while processing /build/src/common/compat_getopt.c.
Error while processing /build/src/common/compat_report_rangecheckfailure.c.
Error while processing /build/src/common/compat___iob_func.c.
Error while processing /build/src/common/simclist.c.
Error while processing /build/src/common/simclist.h.
Error while processing /build/src/common/libpkcs11.c.
Error while processing /build/src/common/libscdl.c.
error: no input files [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat___iob_func.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_dummy.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_getopt.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_getopt.h' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_getpass.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_getpass.h' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_report_rangecheckfailure.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strlcat.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strlcat.h' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strlcpy.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strlcpy.h' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strnlen.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/compat_strnlen.h' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/libpkcs11.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/libscdl.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/simclist.c' [clang-diagnostic-error]
error: no such file or directory: '/build/src/common/simclist.h' [clang-diagnostic-error]
error: unable to handle compilation, expected exactly one compiler job in '' [clang-diagnostic-error]
Found compiler error(s).
make[3]: *** [Makefile:820: check-local] Error 1
make[3]: Leaving directory '/build/src/common'
make[2]: *** [Makefile:671: check-am] Error 2
make[2]: Leaving directory '/build/src/common'
make[1]: *** [Makefile:462: check-recursive] Error 1
make[1]: Leaving directory '/build/src'
make: *** [Makefile:605: check-recursive] Error 1
```
Seems like `TIDY_FILES` (https://github.com/OpenSC/OpenSC/blob/master/src/common/Makefile.am#L33) should use `$(top_srcdir)` to reference them.

--- a/src/tests/unittests/Makefile.am
+++ b/src/tests/unittests/Makefile.am
@@ -12,11 +12,10 @@ TESTS = asn1 simpletlv
 noinst_HEADERS = torture.h
 
 AM_CFLAGS = -I$(top_srcdir)/src/ \
-	-L$(top_srcdir)/src/libopensc/ \
 	$(CODE_COVERAGE_CFLAGS) \
 	$(CMOCKA_CFLAGS)
 AM_CPPFLAGS =$(CODE_COVERAGE_CPPFLAGS)
-LDADD = $(top_srcdir)/src/libopensc/libopensc.la \
+LDADD = $(top_builddir)/src/libopensc/libopensc.la \
 	$(CODE_COVERAGE_LIBS) \
 	$(OPTIONAL_OPENSSL_LIBS) \
 	$(CMOCKA_LIBS)

